adagrams/game.py

```diff
@@ -1,13 +1,137 @@
+    player_hand = []
+    freq_list = []
+
+    # loop to store the key and value of LETTER_POOL to separate list
+    for key in LETTER_POOL:
+        letter_list.append(key)
+        freq_list.append(LETTER_POOL[key])
+
+    #This loop helps to generate a hand of 10 strings and add it to a list
+    for item in range(10):
+        random_index_of_letter_list = randint(0,len(letter_list)-1)
+        letter_picked = letter_list[random_index_of_letter_list]
+        player_hand.append(letter_picked)
+        freq_list[random_index_of_letter_list] -= 1
+        if freq_list[random_index_of_letter_list] == 0:
+            letter_list.pop(random_index_of_letter_list)
+            freq_list.pop(random_index_of_letter_list)
+      
+    return player_hand            
+    
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    """
+    This function checks whether the word passed has characters that are
+    contained within a collection (or hand) of drawn letters
+    input : variable word as a string, list of drawn letters
+    output: True if all characters of the word are present in 
+            the list or else false
+    """
+    
+    word = word.upper() # To convert all the characters to uppercase.
+    letter_bank_copy = letter_bank.copy()
+    for character in word:
+        if character in letter_bank_copy:
+            letter_bank_copy.remove(character)
+        else:
+            return False
+    return True    
 
 def score_word(word):
-    pass
+    """
+    This function calculates the total score of a word.
+    input : word having data type String
+    output: Total score as an integer
+
+    """
+    word = word.upper()
+    sum_of_points = 0
+    for letter in word:
+        for points, list_letter in SCORE_CHART.items():
+            if letter in list_letter:
+                sum_of_points += points
+    if len(word) >= 7:
+        sum_of_points += 8          
+    return sum_of_points
+    
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    """
+    This function calculates the highest score from the 
+    word list and returns the word and the highest score.
+    input: word_list, which is a list of strings
+    output: a tuple that represents the data of a winning word and it's score
+    """
+
+
+    total_score_chart = {}
+
+    # This loop helps to add the word and its score to a dictionary
+    for word in word_list:
+        word_score = score_word(word)
+        total_score_chart[word] = word_score
+
+    max_score = 0 
+    max_word = "" 
+    for word, score in total_score_chart.items():
+        if score > max_score :
+            max_score = score
+            max_word = word
+        elif score == max_score:
+            if len(word) == 10 and len(max_word) != 10:
+                max_word = word
+                max_score = score
+            elif len(word) < len(max_word) and len(max_word) != 10 :
+                max_word = word
+                max_score = score
```

Comment:
I'd prefer the tie breaking logic to be in a helper function to make `get_highest_word_score` more concise and single responsibility. Notice that the name `get_highest_word_score` only indicates getting the highest score (not breaking ties __and__ getting highest score).